ops/terraform/services/eft/lambda_src/sftp_outbound_transfer/sftp_outbound_transfer.py

```diff
@@ -0,0 +1,484 @@
+            sftp_hostname=sftp_hostname,
+            sftp_host_pub_key=sftp_host_pub_key,
+            sftp_username=sftp_username,
+            sftp_user_priv_key=sftp_user_priv_key
+        )
+
+
+@dataclass(frozen=True, eq=True)
+class RecognizedFile:
+    filename_pattern: str
+    destination_folder: str
+
+
+@dataclass(frozen=True, eq=True)
+class PartnerSsmConfig:
+    partner: str
+    bucket_home_dir: str
+    pending_files_dir: str
+    sent_files_dir: str
+    failed_files_dir: str
+    recognized_files: list[RecognizedFile]
+
+    @classmethod
+    def from_partner(cls, partner: str) -> "PartnerSsmConfig":
+        bucket_home_dir = get_ssm_parameter(
+            f"/{partner}/{BFD_ENVIRONMENT}/eft/sensitive/bucket_home_dir", with_decrypt=True
+        )
+        pending_files_dir = get_ssm_parameter(
+            f"/{partner}/{BFD_ENVIRONMENT}/eft/sensitive/outbound/pending_dir", with_decrypt=True
+        )
+        sent_files_dir = get_ssm_parameter(
+            f"/{partner}/{BFD_ENVIRONMENT}/eft/sensitive/outbound/sent_dir", with_decrypt=True
+        )
+        failed_files_dir = get_ssm_parameter(
+            f"/{partner}/{BFD_ENVIRONMENT}/eft/sensitive/outbound/failed_dir", with_decrypt=True
+        )
+        recognized_files = [
+            RecognizedFile(**file_dict)
+            for file_dict in json.loads(
+                get_ssm_parameter(
+                    f"/{partner}/{BFD_ENVIRONMENT}/eft/sensitive/outbound/recognized_files_json",
+                    with_decrypt=True,
+                )
+            )
+        ]
+
+        return PartnerSsmConfig(
+            partner=partner,
+            bucket_home_dir=bucket_home_dir,
+            pending_files_dir=pending_files_dir,
+            sent_files_dir=sent_files_dir,
+            failed_files_dir=failed_files_dir,
+            recognized_files=recognized_files,
+        )
+
+    @property
+    def bucket_home_path(self):
+        return f"{BUCKET_ROOT_DIR}/{self.bucket_home_dir}"
+
+    @property
+    def pending_files_full_path(self):
+        return f"{self.bucket_home_path}/{self.pending_files_dir}"
+
+    @property
+    def sent_files_full_path(self):
+        return f"{self.bucket_home_path}/{self.sent_files_dir}"
+
+    @property
+    def failed_files_full_path(self):
+        return f"{self.bucket_home_path}/{self.failed_files_dir}"
+
+
+def get_ssm_parameter(name: str, with_decrypt: bool = False) -> str:
+    response = ssm_client.get_parameter(Name=name, WithDecryption=with_decrypt)
+
+    try:
+        return response["Parameter"]["Value"]
+    except KeyError as exc:
+        raise ValueError(f'SSM parameter "{name}" not found or empty') from exc
+
+
+def move_s3_object(object_key: str, destination: str) -> None:
+    try:
+        s3_client.copy(
+            CopySource={"Bucket": BUCKET, "Key": object_key},
+            Bucket=BUCKET,
+            Key=destination,
+        )
+        s3_client.delete_object(Bucket=BUCKET, Key=object_key)
```

Comment:
For the unaware: S3 does not support moves, only copies. So, to _simulate_ a move, we first copy the file and then delete it.

Reply:
Is it overly cautious to ensure the object was copied prior to its deletion?

Reply:
My assumption here is that a failure to copy would result in an exception being raised, and we'd exit this function before we try to delete. It's definitely possible to check if the object exists at the new path _before_ deletion and raise a new error if it doesn't, though. That _might_ be redundant, depending on whether or not we trust `boto3` to raise an error for all cases where a copy fails.